[DOCS] Documents imbalanced class sizes and their effect on classification

szabosteve opened the pull request:
This PR adds a limitation item about imbalanced class sizes to the limitation section and mentions the issue in the classification conceptual documentation.

Docs preview: http://stack-docs_806.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-dfa-limitations.html

tveasey commented on docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:
```diff
@@ -49,7 +49,10 @@ means that you need to supply a labeled training dataset that has some
 model on your training dataset, you can reuse the knowledge that the model has 
-learned about the relationships between the data points to classify new data.
+learned about the relationships between the data points to classify new data. 
+Your training dataset should be balanced, otherwise the {classanalysis} may not 
```
This is a bit strong as currently worded IMO, we handle imbalanced training data up to a point
```suggestion
Your training dataset should be approximately balanced, otherwise the {classanalysis} may not 
```

tveasey commented on docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc:
```diff
@@ -104,6 +104,21 @@ where included fields contain an array are also ignored. Documents in the
+If your training data is very imbalanced, then {classanalysis} may not provide 
+good predictions. The system tries to mitigate the effects of imbalanced 
+training data by maximizing the minimum recall of any class. It means that it 
+tries to get correct the highest number of examples in the minority classes. 
```
I think I'd modify these sentences slightly. Here "the system" is really training so I'd probably write something like:
"Training tries to mitigate the effects of imbalanced training data by maximizing the minimum recall of any class. For imbalanced training data, this means that it will try to balance the proportion of values it correctly labels in the minority and majority class."

szabosteve replied:
```suggestion
will try to balance the proportion of values it correctly labels in the minority and majority class.
```

tveasey commented on docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc:
```diff
@@ -104,6 +104,21 @@ where included fields contain an array are also ignored. Documents in the
+training data by maximizing the minimum recall of any class. It means that it 
+tries to get correct the highest number of examples in the minority classes. 
+However, this process can result in a slight degradation of the overall 
+accuracy. Try to avoid highly imbalanced situations. It is required at least 50 
```
I'd weaken these words slightly (and suggest a maximum imbalance), i.e. something like:
"We recommend having at least 50 examples of each class and a ratio of no more than 10 to 1 for the majority to minority class labels in the training data."

tveasey commented on docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc:
```diff
@@ -104,6 +104,21 @@ where included fields contain an array are also ignored. Documents in the
+However, this process can result in a slight degradation of the overall 
+accuracy. Try to avoid highly imbalanced situations. It is required at least 50 
+examples for each class in order to get better results. If your training dataset 
+is very imbalanced, fine-tune the data, for example, by downsampling the 
```
Maybe:
"If your training dataset is very imbalanced, consider downsampling the majority class, upsampling the minority class or if possible gather more data."
(I think it's worth mentioning the three possible routes to go.)

szabosteve commented on docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc:
```diff
@@ -104,6 +104,21 @@ where included fields contain an array are also ignored. Documents in the
+=== Imbalanced class sizes affect {classification} performance
+
+If your training data is very imbalanced, then {classanalysis} may not provide 
+good predictions. The system tries to mitigate the effects of imbalanced 
```
```suggestion
good predictions. Training tries to mitigate the effects of imbalanced 
```

szabosteve commented on docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc:
```diff
@@ -104,6 +104,21 @@ where included fields contain an array are also ignored. Documents in the
+
+If your training data is very imbalanced, then {classanalysis} may not provide 
+good predictions. The system tries to mitigate the effects of imbalanced 
+training data by maximizing the minimum recall of any class. It means that it 
```
```suggestion
training data by maximizing the minimum recall of any class. For imbalanced training data, this means that it
```

szabosteve commented:
@tveasey Thank you for the review. I addressed your feedback and added a defining clause to the sentence in the classification conceptual documentation to explain what "balanced dataset" means.

szabosteve commented:
**Backport commits:**
7.x: ba64d51
7.6: 0f4dbac
7.5: 92eb563
